### PR TITLE
Integrated changes with CD pipeline for deployment and minification

### LINKED ISSUE #299: Navigation from weekly view to daily log
**Describe the bug**
When I click on the column links in weekly view, I am taken to the daily log page. However, the daily log is not populated with the corresponding day.

**To Reproduce**
1. Index page, click on one of the active books 
2. Weekly page, click on the column link
3. See error

**Expected behavior**
The daily log should be populated 

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### LINKED ISSUE #330: WeeklyViewItem sometimes show up sometimes doesn't
**Describe the bug**
When navigate from books to weekly view, the weeklyViewItem show up properly
When navigate directly to weekly view using nav bar, the weeklyViewItem doesn't show up
**To Reproduce**
1. Make a new daily log entry
2. go back to index page
3. click the weekly view from the nav bar
4. see bug
**Expected behavior**
Data persistence is broken here

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### LINKED ISSUE #355: Configure continuous delivery pipeline for deployment and minification
## Detailed Description

Since our final sprint is almost finished, we are primarily interested in revisiting our CD pipeline for deployment. We wish that the following workflow is respected in our CD pipeline:
1. PR is made against the `main` branch.
2. Once changes are approved, we run a job that does the following:
  i. Minify all JavaScript, CSS, and HTML files
  ii. Run all unit tests
  iii. Push changes beneath the `source` directory to a branch titled `gh-pages`
3. Once `gh-pages` is updated, we should use Github pages to deploy this branch (this will serve as our production branch).

## Context

Since time permits, having a more refined CD pipeline (removing overhead from `node_modules`, minifying files, etc.) ensures that our site deployed to production has as little overhead as possible (this is closely related to the RAIL model).

## Possible Implementation

[This github action allows us to deploy a subdirectory of `main` to Github pages](https://github.com/maxheld83/ghpages). With regards to minification, we need to look into a task runner like [Gulp](https://gulpjs.com/).

## Your Environment

* Browser Name and version:--
* Operating System and version (desktop or mobile):--

### PR DESCRIPTION
**Pull request by NoahT**
For our CD pipeline, we were primarily interested in deploying to Github pages and minifying all relevant assets once pull requests are merged into `main`. This pull request reflects the necessary changes to our CD pipeline so that our deployment process operates in this manner. Resolves #355

### CODE REVIEW

**Comment by NoahT** [comment]
@yuzilyu @will-chung don't be alarmed if linting fails; recall in our CI jobs for testing we build our application prior to linting. Since our build process now includes minification, we can just change this section of the `node.js.yml` file on a different PR

File: source/javascript/src/daily.js [hunk 1]
```diff
@@ -443,7 +445,15 @@ document.addEventListener('DOMContentLoaded', (event) => {
   time.addEventListener('change', () => {
     date.style.visibility = 'visible'
   })
-  getLogInfoAsJSON(populateDailyLog)
+
+  fetch('https://api.quotable.io/random')
```
Comment (will-chung) [inline source/javascript/src/daily.js:449]: Quotes FTW 😄

Comment (NoahT) [reply source/javascript/src/daily.js:449]: Good catch; We should probably add the new ReST API calls to our architecture diagrams as well. @yuzilyu what do you think?

**Review by will-chung — APPROVED** [review]
Great changes, everything looks good!

File: source/javascript/src/daily.js (hunk 1 above)
Comment (yuzilyu) [reply source/javascript/src/daily.js:449]: Sounds good!

**Review by yuzilyu — APPROVED** [review]

**Pull request merged**